dcos/cosmos.py

```diff
@@ -180,7 +180,8 @@ def _cosmos_request(self,
                         response.headers.get('Content-Type')))
             return response
         except DCOSBadRequest as e:
-            if len(headers_preference) > 1:
+            error_type = e.response.json().get('type')
+            if error_type == 'not_valid' and len(headers_preference) > 1:
```

Comment:
This type of errors are constructed here: https://github.com/dcos/cosmos/blob/master/cosmos-common/src/main/scala/com/mesosphere/cosmos/circe/Encoders.scala#L37-L52